server/controllers/stock/import.js

```diff
@@ -34,79 +34,98 @@ function downloadTemplate(req, res, next) {
-      });
-
-      const isExit = 0;
-      const postingParams = [documentUuid, isExit, req.session.project.id, req.session.enterprise.currency_id];
-
-      if (req.session.stock_settings.enable_auto_stock_accounting) {
-        transaction.addQuery('CALL PostStockMovement(?)', [postingParams]);
-      }
-
-      return transaction.execute();
-    })
-    .then(() => res.sendStatus(201))
-    .catch(next)
-    .done();
+
+  try {
+    // check if a depot exists for the given uuid
+    await db.one('SELECT uuid FROM depot WHERE uuid = ?', depotUuid);
+
+    // get the fiscal year period information
+    const period = await Fiscal.lookupFiscalYearByDate(new Date());
+
+    // read the csv file
+    const data = await util.formatCsvToJson(filePath);
+
+    // check validity of all data from the csv file
+    checkDataFormat(data);
+
+    const transaction = db.transaction();
+    const query = 'CALL ImportStock(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);';
+
+    data.forEach(item => {
+      queryParams = [
+        req.session.enterprise.id,
+        req.session.project.id,
+        req.session.user.id,
+        depotUuid,
+        documentUuid,
+        item.inventory_group_name,
+        item.inventory_code || '',
+        item.inventory_text,
+        item.inventory_type,
+        item.inventory_unit,
+        item.inventory_unit_price,
+        item.inventory_cmm,
+        item.stock_lot_label,
+        item.stock_lot_quantity,
+        moment(item.stock_lot_expiration).format('YYYY-MM-DD'),
+        period.id,
+      ];
+      transaction.addQuery(query, queryParams);
+    });
+
+    const isExit = 0;
+    const postingParams = [documentUuid, isExit, req.session.project.id, req.session.enterprise.currency_id];
+
+    if (req.session.stock_settings.enable_auto_stock_accounting) {
+      transaction.addQuery('CALL PostStockMovement(?)', [postingParams]);
+    }
+
+    await transaction.execute();
+    res.sendStatus(201);
+  } catch (error) {
+    next(error);
+  }
 }
 
 /**
- * hasValidDataFormat
+ * checkDataFormat
  *
  * @description check if data has a valid format for stock
  *
  * @param {object} data
  */
-function hasValidDataFormat(data = []) {
-  return data.every(item => {
+function checkDataFormat(data = []) {
+  for (let i = 0; i < data.length; i++) {
+    const item = data[i];
     const isUnitPriceDefined = typeof (item.inventory_unit_price) !== 'undefined';
-    return item.inventory_group_name
+    const isUnitPriceNumber = !Number.isNaN(Number(item.inventory_unit_price));
+    const isLotQuantityNumber = !Number.isNaN(Number(item.stock_lot_quantity));
+    const validity = item.inventory_group_name
       && item.inventory_text && item.inventory_type && item.inventory_unit
       && isUnitPriceDefined && item.stock_lot_label
-      && item.stock_lot_quantity && item.stock_lot_expiration;
-  });
+      && item.stock_lot_quantity && item.stock_lot_expiration
+      && isUnitPriceNumber && isLotQuantityNumber;
+
```

Comment:
You should consider added more checks for each field:  checking that the text fields have text, that numbers have numbers (such as stock_lot_quantity, etc).  I know that writing the code to check each item is a bit tedious, but it will be a tremendous help to users down the road when you tell them exactly what is wrong with their file (instead of "bad data format").

Reply:
This may be helpful: https://github.com/validatorjs/validator.js